### PR TITLE
Usability improvements in the LUKS decrypt dialog

### PR DESCRIPTION
**Pull request by ancorgs**
## Problem

The problem is already described at #165, the dialog for decrypting a LUKS device has several usability problems including:

- The wording is a bit weird (kind of inconsistent with the labels of the buttons).
- Clicking "enter" does not submit the password, instead it causes a page reload.
- The default option is to skip decryption which:
  - Hurts usability with keyboard (and you are already using the keyboard if you entered the password, isn't it?)
  - Makes too easy to skip decryption even after you have already entered a password

## Solution

- Simpler wording, based on the YaST one (see YaST screenshot at #165)
- "Decrypt" is enforced to be the default option.
- Ensure pressing "enter" works as expected, that is:
  - Doing nothing if there is no password in the corresponding input (since the "Decrypt" button is disabled).
  - Submitting the entered password if there is any written.

## Still pending

The buttons don't look like in the other dialogs. They are aligned to the left.

## Potentially controversial code changes

- type=submit
- hardcoded string

## Testing

Tested manually

## Screenshots

decrypt

### CODE REVIEW

**Comment by ancorgs** [comment]
After performing all these changes (eg. putting the buttons inside the form) I realized the enter key works properly in the "New root password" popup. I want to check what the difference because it could mean there is a less intrusive way to fix the original problem in the decryption dialog.

**Comment by dgdavid** [comment]
> After performing all these changes (eg. putting the buttons inside the form) I realized the enter key works properly in the "New root password" popup. I want to check what the difference because it could mean there is a less intrusive way to fix the original problem in the decryption dialog.

Excuse me @ancorgs . Blame to my bad memory. I completely forgot that @joseivanlopez and me were working on the implicit submission issue  few months ago and solved it by using the [`form` attribute](https://html.spec.whatwg.org/#the-button-element) in buttons to avoid the implicit submission. Indeed, it was the less intrusive solution we found.

**Comment by dgdavid** [comment]
> using the [`form` attribute](https://html.spec.whatwg.org/#the-button-element) in buttons to avoid the implicit submission

See https://github.com/yast/d-installer/commit/45d3efc4bc6ee389fc5596e9fc6c5802d5c602bb

**Comment by dgdavid** [comment]
> The buttons don't look like in the other dialogs. They are aligned to the left.

**TL;DR**

This is because these buttons are no longer placed in the Modal footer after you nested them inside an `ActionGroup`.

**Long Story**

To follow EOS recommendations and place modal buttons at the right, we simply put a `flex-direction: row-reverse` CSS rule

https://github.com/yast/d-installer/blob/66c1bec8a926b34dd1010da3922c3ddac4b88c5d/web/src/patternfly.scss#L93-L96

It is still working, but after you nested `Popup.Actions` the `Popup` component is not able to extract them using the `React.Children` API because it is no longer a direct child of the Popup.

https://github.com/yast/d-installer/blob/66c1bec8a926b34dd1010da3922c3ddac4b88c5d/web/src/Popup.jsx#L196-L198

I.e., `actions` is an empty array now and these buttons are not placed in the [`PatternFly Modal`](https://www.patternfly.org/v4/components/modal)  _\<footer\>_ but in its body. Something to document better or even to improve, I guess.

**Comment by ancorgs** [comment]
> [...] I realized the enter key works properly in the "New root password" popup. I want to check what the difference because it could mean there is a less intrusive way to fix the original problem in the decryption dialog.

After checking how things are done there, I came up with #253 which looks to me like a less intrusive change (and more consistent with other dialogs in the UI).

I'm closing this pull request in favor of that other one.
